### PR TITLE
fix: set default selectedNetworkClientId to 'mainnet' if no matching with entry on network list

### LINKED ISSUE #11657: [Sentry] Error: Engine does not exist
Sentry Issue: [METAMASK-MOBILE-2QW9](https://metamask.sentry.io/issues/5837039953/?referrer=github_integration)

```
Error: Engine does not exist
  at assertEngineExists (app/core/Engine.ts:1948:20)
  at get context (app/core/Engine.ts:1956:23)
  at ?anon_0_ (app/core/Authentication/Authentication.ts:468:46)
  at ?anon_0_ (app/core/Authentication/Authentication.ts:453:41)
...
(22 additional frame(s) were not displayed)
```

This was fixed in June in #9958 but seems to be back

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**
This PR implements a fix for Migration 60 to handle cases where selectedNetworkClientId does not match any entry within networkConfigurationsByChainId. If no corresponding networkClientId is found, the migration will now set selectedNetworkClientId to 'mainnet' by default.

## **Related issues**

Fixes: [#11657](https://github.com/MetaMask/metamask-mobile/issues/11657)

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/store/migrations/062.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    !hasProperty(state.engine, 'backgroundState')
+  ) {
+    captureException(
+      new Error(
+        `Migration 62: Invalid or missing 'backgroundState' in engine: '${typeof state
+          .engine.backgroundState}'`,
+      ),
+    );
+    return state;
+  }
```
Comment (Cal-L) [inline app/store/migrations/062.ts:40]: You can replace everything above this with `ensureValidState`

File: app/store/migrations/062.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  ) {
+    captureException(
+      new Error(
+        `Migration 62: Invalid or missing 'NetworkController' in backgroundState: '${typeof networkControllerState}'`,
+      ),
+    );
+    return state;
+  }
+
+  if (
```
Comment (Cal-L) [inline app/store/migrations/062.ts:56]: This condition will prematurely throw if `selectedNetworkClientId` is in a bad state. Do we still want this condition if we can potentially resolve this by setting the default value to `mainnet` instead?

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/store/migrations/062.ts (hunk 2 above)
Comment (salimtb) [reply app/store/migrations/062.ts:56]: aah yes i see what you mean , we can indeed resolve it to mainnet by default

**Review by Cal-L — APPROVED** [review]
Lgtm

**Pull request merged**